pkg/irreparable/irreparable_test.go

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package irreparable
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+	"storj.io/storj/internal/testcontext"
+)
+
+func TestIrreparableRepository(t *testing.T) {
+	//testing variables
+	var (
+		rmtsegpath         = []byte("IamSegmentkeyinfo")
+		rmtsegdetail       = []byte("IamSegmentdetailinfo")
+		piecesLost         = int64(10)
+		damagedsegUnixsec  = time.Now().Unix()
+		repairAttemptCount = int64(10)
+	)
+
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	// creating in-memory db and opening connection
+	irrdb, err := newTestIrreparableDB()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer ctx.Check(irrdb.DB.Close)
+
+	t.Run("New Entry", func(t *testing.T) {
+		rmtseginfo := &RemoteSegmentInfo{
+			SegmentPath:        rmtsegpath,
+			SegmentDetail:      rmtsegdetail,
+			LostPiecesCount:    piecesLost,
+			RepairUnixSec:      damagedsegUnixsec,
+			RepairAttemptCount: repairAttemptCount,
+		}
+
+		err = irrdb.Put(ctx, rmtseginfo)
+		assert.NoError(t, err)
+	})
+
+	t.Run("Entry already exists", func(t *testing.T) {
+		createReq := &RemoteSegmentInfo{
+			SegmentPath:        rmtsegpath,
+			SegmentDetail:      rmtsegdetail,
+			LostPiecesCount:    piecesLost,
+			RepairUnixSec:      damagedsegUnixsec,
+			RepairAttemptCount: repairAttemptCount,
+		}
+
+		err = irrdb.Put(ctx, createReq)
+		assert.NoError(t, err)
+
+		dbrmtsegInfo, err := irrdb.Get(ctx, createReq)
+		assert.NoError(t, err)
+		assert.EqualValues(t, rmtsegpath, dbrmtsegInfo.SegmentPath, rmtsegpath)
+		assert.EqualValues(t, rmtsegdetail, dbrmtsegInfo.SegmentDetail, rmtsegdetail)
+		assert.EqualValues(t, piecesLost, dbrmtsegInfo.LostPiecesCount, piecesLost)
+		assert.EqualValues(t, damagedsegUnixsec, dbrmtsegInfo.RepairUnixSec, damagedsegUnixsec)
+		assert.EqualValues(t, repairAttemptCount, dbrmtsegInfo.RepairAttemptCount, repairAttemptCount)
+	})
+
+	t.Run("Delete existing entry", func(t *testing.T) {
+		delReq := &RemoteSegmentInfo{
+			SegmentPath:        rmtsegpath,
+			SegmentDetail:      rmtsegdetail,
+			LostPiecesCount:    piecesLost,
+			RepairUnixSec:      damagedsegUnixsec,
+			RepairAttemptCount: repairAttemptCount,
+		}
+
+		err = irrdb.Delete(ctx, delReq)
+		assert.NoError(t, err)
+
+		_, err := irrdb.Get(ctx, delReq)
+		assert.Error(t, err)
+	})
+}
+
+func newTestIrreparableDB() (s *Database, err error) {
```

Comment:
You really don't need this func you can call above
```
db, err := New("sqlite://memory")
```

Reply:
Done